src/index.ts

```diff
@@ -1003,6 +1003,53 @@ export abstract class PinejsClientCore<PinejsClient> {
 		return this.request({ ...params, method: 'DELETE' });
 	}
 
+	public async getOrCreate(params: GetOrCreateParams) {
+		const { id, body, ...restParams } = params;
+
+		if (params.resource.endsWith('/$count')) {
+			throw new Error('getOrCreate does not support $count on resources');
+		}
+
+		if (body == null) {
+			throw new Error('The body property is missing');
+		}
+
+		let uniqueFields: Dictionary<any>;
+		if (isPrimitive(id) || (isObject(id) && '@' in id)) {
+			uniqueFields = {
+				id,
+			};
```

Comment:
This is a little sketchy as it's not guaranteed to be true, eg it's possible to have a primary key that is not named `id`, overall it probably makes more sense to require the explicit named fields version because in that case it can still be `id = { id: 1 }` if that's what's necessary